# Chapter 2 - The Basic Theory of Interest

## Exercise 1.

**(a)** Suppose we invested $1 in 1776 at 3.3% interest compounded yearly and we assume the current date to be 2009 (the date of publication), then we have for $r = 0.033$ and $n = (2009 - 1) - (1776 + 1) = 231$ years. This calculation allows for the yearly interest at the start of 2009 and accounts for the fact that interest only begins to accrue from 1777. Thus we have the present value of the investment given by

$$
\begin{equation*}
    \begin{split}
        \text{PV} &= {(1 + r)}^n \\
                  &= {1.033}^{231} \\
                  &\approx 1807.90.
    \end{split}
\end{equation*}
$$

We use the following to calculate the approximate value:

In [79]:
pv = 1.033**231
print(pv)

1807.899443302222


The closest estimate is thus $1,000.

**(b)** If the interest rate were 6.6% compounded annually, we would have the following present value

$$
\begin{equation*}
    \begin{split}
        \text{PV} &= {(1 + r)}^n \\
                  &= {1.066}^{231} \\
                  &\approx 2581750.58.
    \end{split}
\end{equation*}
$$

In [80]:
pv = 1.066**231
print(pv)

2581750.581539662


Thus the new approximation would be $2,581,750.58.

## Exercise 2.

We have 

$$
\begin{equation*}
    \begin{split}
                             {(1 + r)}^n &= 2 \\
        \Leftrightarrow \ln{{(1 + r)}^n} &= \ln{2} \\ 
           \Leftrightarrow n\ln{(1 + r)} &= \ln{2} \\
                       \Leftrightarrow n &= \frac{\ln{2}}{\ln{(1 + r)}}.
    \end{split}
\end{equation*}
$$

Then, applying the approximation $ln{(1 + r)} \approx r$ for small $r$, we have

$$
\begin{equation*}
    \begin{split}
        n &\approx \frac{\ln{2}}{r} \\
          &= \frac{.69}{r}.
    \end{split}
\end{equation*}
$$

Thus, using $i = 100r$, we have

$$
\begin{equation*}
    \begin{split}
        n &\approx \frac{\ln{2}}{r} \\
          &= \frac{.69}{(i / 100)} \\
          &= \frac{69}{i}.
    \end{split}
\end{equation*}
$$

Now, using the better approximation $\ln{(1 + r)} \approx r - \frac{1}{2}r^2$, we have for $r \approx .08$ we conclude that

$$
\begin{equation*}
    \begin{split}
        n &\approx \frac{\ln{2}}{r - \frac{1}{2}r^2} \\
          &= \frac{.69}{.08 - \frac{1}{2}{.08}^2}.
    \end{split}
\end{equation*}
$$

Using the following calculation

In [81]:
n = .69 / (.08 - (.5 * (.08**2)))
print(n)

8.984374999999998


For $72 / i = 72 / 100r \approx 72 / (100 \cdot .08)$, we have the following calculation

In [82]:
n_approx = 72 / (100 * .08)
print(n_approx)
print(abs(n_approx - n))


9.0
0.015625000000001776


Thus we can see that $n \approx 72 / i$.


## Exercise 3.

**(a)** An annual rate of 3% compounded monthly will produce an increase of ${(1 + r)}^n = {(1.0025)}^{12}$, which we approximate by the following

In [83]:
increase = (1.0025 ** 12)
print(increase)

1.0304159569135067


Thus the effective rate is given by 3.04%.

**(b)** An annual rate of 18% compounded monthly will produce an increase of ${(1.015)}^{12}$, which we approximate by the following

In [84]:
increase = (1.015 ** 12)
print(increase)

1.195618171461534


Thus the effective rate is given by 19.56%.

**(c)** An annual rate of 18% compounded quarterly will produce an increase of ${(1.045)}^4$, which we approximate by the following

In [85]:
increase = (1.045 ** 4)
print(increase)

1.1925186006249997


Thus the effective rate is given by 19.25%.

## Exercise 4.

We have the following definition of Newton's method and the appropriate functions:

In [86]:
from typing import Callable


def newtons_method(f: Callable[[float], float], f_p: Callable[[float], float], lambda_k: float) -> float:
    return lambda_k - (f(lambda_k) / f_p(lambda_k))


lambdas: list[float] = [1.0]


def f(x: float) -> float:
    return -1.0 + x + x**2


def f_p(x: float) -> float:
    return 1.0 + 2.0 * x


for k in range(1, 5):
    lambdas.append(newtons_method(f, f_p, lambdas[k - 1]))


for lambda_bar in lambdas:
    print(f"lambda_bar = {lambda_bar}, f(lambda_bar) = {f(lambda_bar)}")

lambda_bar = 1.0, f(lambda_bar) = 1.0
lambda_bar = 0.6666666666666667, f(lambda_bar) = 0.11111111111111127
lambda_bar = 0.6190476190476191, f(lambda_bar) = 0.0022675736961451642
lambda_bar = 0.6180344478216818, f(lambda_bar) = 1.0265159329780893e-06
lambda_bar = 0.618033988749989, f(lambda_bar) = 2.1066481892262345e-13


Thus we conclude that the approximation after four estimates is $\bar{\lambda} \approx 0.618033988749989$ and $f(\bar{\lambda}) \approx 2.1066481892262345 \times {10}^{-13} \approx 0$.

## Exercise 5.

Given payments $x_k = 500000$, for $k = 0, 1, \ldots, 20$, and $r = 0.1$, we have the following present value: 

$$
\begin{equation*}
    \begin{split}
        \text{PV} &= \sum_{k = 0}^n{\frac{x_k}{{(1 + r)}^k}} \\
                  &= \sum_{k = 0}^{20}{\frac{500000}{1.1^k}} \\
                  &\approx 4756781.86,
    \end{split}
\end{equation*}
$$

so that the present value of the prize is $4,756,781.86.

We use the following to calculate the approximate present value:

In [87]:
n = 20
r = 0.1
pv = 0.0
x_k = 500000
for k in range(0, n + 1):
    pv += x_k / ((1 + r) ** k)

print(pv)

4756781.85987928


## Exercise 6.

Assuming that the couple wants to stay in the apartment for 6 months, we have two cash flow streams $\mathbf{x} = (1000, 1000, 1000, 1000, 1000, 1000)$ and $\mathbf{z} = (1000, 900, 900, 900, 900, 900)$, with $r = 0.12$, and $k = 0, 1, \ldots, 6 = n$, so that we have the following present values:

$$
\begin{equation*}
    \begin{split}
        {\text{PV}}_{\mathbf{x}} &= \sum_{k = 0}^n{\frac{x_k}{{(1 + r)}^k}} \\
                                 &= \sum_{k = 0}^6{\frac{1000}{1.12^k}} \\
                                 &\approx 5111.41,
    \end{split}
\end{equation*}
$$

and

$$
\begin{equation*}
    \begin{split}
        {\text{PV}}_{\mathbf{z}} &= \sum_{k = 0}^n{\frac{z_k}{{(1 + r)}^k}} \\
                                 &= 1000 + \sum_{k = 1}^6{\frac{900}{1.12^k}} \\
                                 &\approx 4700.27,
    \end{split}
\end{equation*}
$$

so that the couple should choose to move to the cheaper apartment.

We use the following to calculate the approximate present values:

In [88]:
n = 6
r = 0.12

pv_x = 0.0
xs = [1000] * (n + 1)
for k in range(0, n + 1):
    pv_x += xs[k] / ((1 + r) ** k)

print(pv_x)

pv_z = 0.0
zs = [1000] + ([900] * (n + 1))
for k in range(0, n + 1):
    pv_z += zs[k] / ((1 + r) ** k)

print(pv_z)

5111.407323522325
4700.2665911700915


Assuming that the couple wants to stay for 1 year, we would have $n = 12$, with ${\text{PV}}_{\mathbf{x}} \approx \$7,194.37$ and $\text{PV}_{\mathbf{z}} \approx \$6,574.94$, so that again the couple should choose to move to the cheaper apartment. 

We use the following to calculate the approximate present values:

In [89]:
n = 12
r = 0.12

pv_x = 0.0
xs = [1000] * (n + 1)
for k in range(0, n + 1):
    pv_x += xs[k] / ((1 + r) ** k)

print(pv_x)

pv_z = 0.0
zs = [1000] + ([900] * (n + 1))
for k in range(0, n + 1):
    pv_z += zs[k] / ((1 + r) ** k)

print(pv_z)

7194.374225455087
6574.936802909578


## Exercise 7

Delaying the harvest another year with the same prevailing interest rate of 10% and the return $R$ would yield the formula

$$
    \text{NPV} = -1 + R / {(1.1)}^2 < 1.48,
$$

so that we have

$$
\begin{equation*}
    \begin{split}
            R / {(1.1)}^2 &< 2.48 \\
        \Leftrightarrow R &< {(1.1)}^2 \cdot 2.48 \\
                          &= 2.904,
    \end{split}
\end{equation*}
$$

so that $x = 2.904$.

## Exercise 8

